### PR TITLE
Invoke new document API for fully load in code broom.

### PR DESCRIPTION
**Pull request by dibarbet**
Related to #34133

Wait until solution is fully loaded when the user hits the code broom.
code_broom_wait

### CODE REVIEW

**Comment by heejaechang** [comment]
bug opened for wait dialog window issue - https://devdiv.visualstudio.com/DevDiv/_workitems/edit/823051

shell team will take a look to see whether they can make the window to show more than 3 lines of description in 16.1

File: src/VisualStudio/CSharp/Impl/LanguageService/CSharpCodeCleanupFixer.cs [hunk 1]
```diff
@@ -309,6 +311,11 @@ private async Task<bool> FixTextBufferAsync(TextBufferCodeCleanUpScope textBuffe
                 cancellationToken = cancellationTokenSource.Token;
 
                 var buffer = textBufferScope.SubjectBuffer;
+
+                // Load document here so that the partial load message actually shows up before the next scope kicks in.
+                var document = buffer.CurrentSnapshot.GetFullyLoadedOpenDocumentInCurrentContextWithChangesAsync(
+                        context.OperationContext).WaitAndGetResult(cancellationToken);
```
Comment (sharwell) [inline src/VisualStudio/CSharp/Impl/LanguageService/CSharpCodeCleanupFixer.cs]: ❗️ Do not call `WaitAndGetResult` from an asynchronous method. This should use `await` instead.

File: src/VisualStudio/CSharp/Impl/LanguageService/CSharpCodeCleanupFixer.cs [hunk 2]
```diff
@@ -309,6 +311,10 @@ private async Task<bool> FixTextBufferAsync(TextBufferCodeCleanUpScope textBuffe
                 cancellationToken = cancellationTokenSource.Token;
 
                 var buffer = textBufferScope.SubjectBuffer;
+
+                // Load document here so that the partial load message actually shows up before the next scope kicks in.
+                var document = await buffer.CurrentSnapshot.GetFullyLoadedOpenDocumentInCurrentContextWithChangesAsync(context.OperationContext).ConfigureAwait(false);
```
Comment (sharwell) [inline src/VisualStudio/CSharp/Impl/LanguageService/CSharpCodeCleanupFixer.cs]: :exclamation: This method uses JTF for switching contexts (line 353, 355), so it needs to use `ConfigureAwait(true)`

```suggestion
                var document = await buffer.CurrentSnapshot.GetFullyLoadedOpenDocumentInCurrentContextWithChangesAsync(context.OperationContext).ConfigureAwait(true);
```

**Comment by dibarbet** [comment]
Discarding this review.  Merge conflicts entirely change how I have to handle this.

**Pull request closed without merging**